docs/project/requirements/architecture.md

```diff
@@ -0,0 +1,147 @@
+# Architecture
+
+Un schéma d'implémentation détaillé sera établi pour décrire l'architecture globale de l'infrastructure, y compris les différents composants, leur interaction et leur déploiement.
+
+
+## Application
+
+
+### Proxy: Traefik
+
+* Application router
+* High-availability by fault tolerance
+* Load-balancing by requests distribution
+  ALB ?
+* Helm chart
+* Version
+
+
+### Application: FastAPI
+
+* Python
+  * Version:
+  * Packaging
+* Project management? Poetry?
+
+
+### Databases: PostgreSQL
+
+* Helm chart
+* Version
+* Master/slave for high availability and read load balancing
+* Master failover
+
+
+## Infrastructure
+
+
+### Network
+
+
+#### Application Load Balancer (ALB)
+
+
+#### Virtual Private Cloud (VPC)
+
+* VPC subnet
+
+
+#### Internet Gateways (IGW)
+
+* Global egress
+
+
+#### Availability Zones (AZ)
+
+* 2 availability zones for high availability
+
+
+#### NAT Gateways (NGW)
+
+* Per AZ egress
+* In public subnet
+
+
+#### Subnets
+
+
+##### Public subnets
+
+* Application access
+* Bastion access
+* NAT gateways ?
+
+
+##### Private subnets
+
+* Database
+* Admin
+
+
+### Compute
+
+#### Nodes
+
+* EC2 compute nodes
+  * Application nodes
+  * Database nodes
+  * Bastion node
+* Auto-scaling groups
+
+
+#### Orchestration: Kubernetes
+
+#### Pods
+
+##### ReplicaSets
+
+* Traefik
+* Application
+* Database
+* cert-manager
+* Metrics server
+* bastion
+
+##### DaemonSets
+
+* Any ?
+
+
+### Storage
+
+#### CSI
+
+##### EBS
+
+* FIXME
+
+
+## CI/CD
+
+### GitHub Actions
+
+
+#### Actions
+
+#### Workflows
+
+* Application
+* Infrastructure
+
+
+## Observability
+
+### Log management (ELK/EFK)
+
+### Metrics (Prometheus/Grafana)
```

Comment:
We use metrics from the following sources:
- kubernetes (comes preinstalled with the kube-prometheus-stack helm package)
- postgres exporter, which runs as sidecars along our postgres clusters
- from our fastapi app using the Instrumentator

Reply:
Does the kube-prometheus-stack collects metrics about nodes?

What would be metrics monitoring black holes at this point?
- EC2/Nodes
- ELB/Traefick?
- EBS ?

Reply:
There's metrics collection happening about the nodes inside the cluster. Additional monitoring outside of the cluster could be useful (eg. EBS utilisation, k8s cluster health, etc.).
We have very little customisation to the default kube-prometheus-stack installation.

Reply:
You mean that kube monitoring stack doesn't watch k8s cluster health?!

I believe that a solid list of metrics (families or targets) that are or should be implemented is in order to show that we know where we have to keep our eyes.